framework/Shell/Actions/TWebServerAction.php

```diff
@@ -0,0 +1,374 @@
+ * if the machine "hosts" file maps a domain name to 127.0.0.1/localhost, then that
+ * domain name can be specified with `--address=testdomain.com` and is accessible
+ * on the machine web browser at "http://testdomain.com/".  In this example, testdomain.com
+ * maps to 127.0.0.1 in the system's "hosts" file.
+ *
+ * The network address can be changed to IPv6 with the command line option `--ipv6`.
+ * To serve pages on all network addresses, including any internet IP address, include
+ * the option `--all`.  These options only work when there is no address specified.
+ *
+ * The command line option `--port=8777` can be used to change the port; in this example
+ * to port 8777.  Ports 1023 and below are typically reserved for application and
+ * system use and cannot be specified without administration access.
+ *
+ * To have more than one worker (to handle multiple requests), specify the `--workers=8`
+ * command option (with the number of page workers need for you).  In this example,
+ * eight concurrent workers are created.
+ *
+ * The TWebServerAction is only available when the application is in "Debug" mode.
+ * In other Application modes, this action can be enabled with the Prado Application
+ * Parameter "Prado:PhpWebServer" set to "true". eg. Within the Application configuration:
+ *
+ * ```xml
+ * <parameters>
+ *     <parameter id="Prado:PhpWebServer" value="true" />
+ * </parameters>
+ * ```
+ *
+ * @author Brad Anderson <belisoful@icloud.com>
+ * @since 4.2.3
+ */
+class TWebServerAction extends TShellAction
+{
+	public const DEV_WEBSERVER_ENV = 'PRADO_DEV_WEBSERVER';
+	public const WORKERS_ENV = 'PHP_CLI_SERVER_WORKERS';
+
+	public const DEV_WEBSERVER_PARAM = 'Prado:PhpWebServer';
+
+	protected $action = 'http';
+	protected $methods = ['serve'];
+	protected $parameters = [[]];
+	protected $optional = [['router-filepath']];
+	protected $description = [
+		'Provides a PHP Web Server to serve the application.',
+		'Runs a PHP Web Server after Initializing the Application.'];
+
+	/** @var bool Listen on all network addresses assigned to the computer, when one is not provided. */
+	private bool $_all = false;
+
+	/** @var ?string The specific address to listen on. */
+	private ?string $_address = null;
+
+	/** @var int The port to listen on, default 8080 */
+	private int $_port = 8080;
+
+	/** @var bool Use a direct ip v6 address, when one is not provided. */
+	private bool $_ipv6 = false;
+
+	/** @var int the number of workers for the Web Server */
+	private int $_workers = 1;
+
+
+
+	/**
+	 * This option is only used when no network interface is specified.
+	 * @return bool Respond on all network addresses.
+	 */
+	public function getAll(): bool
+	{
+		return $this->_all;
+	}
+
+	/**
+	 * This option is only used when no network interface is specified.
+	 * @param mixed $value
+	 * @return bool Respond on all network addresses.
+	 * @return static The current object.
+	 */
+	public function setAll($value): static
+	{
+		if (!$value) {
```

Comment:
I guess you want to check is `$value === null`, otherway a call to `setAll(false)` will set `$this->_all` to true.

Reply:
Good point.  I'll correct this.